### DISCUSSION TITLE
Retrieve Source from Org doesn't work for folders (SFDX menu)

### DISCUSSION THREAD

**Discussion by Dovbush** [forcedotcom/salesforcedx-vscode#5751] [Ideas]
Hi guys,

### Summary

When you right click on the folder to retrieve it's content from org - then it doesn't retrieve new components. Example: you have aura folder with existing components, if you create new Aura cmp outside VS Code (Dev console, another IDE) and then go back to VS Code project, right click on aura folder and select retrieve from org - new component is not retrieved.

### Steps To Reproduce:

1.  Open existing SF projects in VS code.
2. Open package.xml and make sure type StaticResource is included (* members).
3. In package.xml right click and select command SFDX: Retrieve Source in Manifest from Org. Wait until is successfully completed.
4.  Open same SF org in browser.
5. Go to Setup -> Develop -> Static Resources -> click New -> Provide dummy name (testRecource), file, select public -> Save.
6. Go back to VS code project - in file tree find folder 'staticresources' -> right click -> select SFDX: Retrieve Source from Org -> wait until is successfully completed.

### Expected result

Newly created static resource 'testRecource' (created in browser) was retrieved in VS code project

### Actual result

Newly created static resource 'testRecource' (created in browser) was NOT retrieved in VS code project, even when clicking refresh wheel on top of files tree.
I also checked static resources directory and there is no new file.

### Additional information

To retrieve new component you need to execute Retrieve command in package.xml and it fetch new component from org - but it takes time.

**VS Code Version**: Version 1.36.1 (1.36.1)

**SFDX CLI Version**: 46.4.0

**OS and version**: mac OS Mojave 10.14.5

**Comment by arohitu** [comment]
@Dovbush Any solution to this issue?

**Comment by Dovbush** [comment]
@arohitu I've just tried again and still can reproduce the issue. I created new apex class in dev console, then in vs code I clicked Retreive Source from Org on 'classes' folder I couldn't find new class retreived.
As a WORKAROUND you can open package.xml file of your project and execute Retreive Source in Manifest from Org - it retrieved my new class (and all other new components) - it just takes a bit longer that separate command on a dedicated folder.

**Comment by git2gus** [comment]
This issue has been linked to a new work item: W-12285249

**Comment by klewis-sfdc** [comment]
Thanks @arohitu and @Dovbush, another workaround would be use the cli directly with force:source:retrieve and the metadata type: `sfdx force:source:retrieve -m ApexClass`.  VS Code should replicate that behavior.  I've created a WI so that we can get this updated.

**Comment by klewis-sfdc** [comment]
Hi @arohitu and @Dovbush,  
Since project folders can be renamed and/or can contain various types of metadata, I think the best path forward here is to add a new command-palette command that can be used to retrieve a specific metadata type.  This would provide the equivalent of `sfdx force:source:retrieve -m ApexClass` in VS Code.  I have spoken with @AnanyaJha about this and we will look at adding the new command as a resolution for this issue.  Any feedback that you have as we move forward with that implementation would be greatly appreciated! -Ken
